core/src/avm1/globals/file_reference.rs

```diff
@@ -21,13 +21,13 @@ use url::Url;
 
 #[derive(Clone, Copy, Collect)]
 #[collect(no_drop)]
-pub struct FileReferenceObject<'gc>(GcCell<'gc, FileReferenceData<'gc>>);
+pub struct FileReferenceObject<'gc>(pub(crate) GcCell<'gc, FileReferenceData<'gc>>);
 
 impl<'gc> FileReferenceObject<'gc> {
     pub fn init_from_dialog_result(
         &self,
         activation: &mut Activation<'_, 'gc>,
-        dialog_result: &dyn FileDialogResult,
+        dialog_result: &dyn FileSelection,
```

Comment:
Since the parameter is now `FileSelection`, and not `FileDialogResult`, I think the name of both the function and the parameter is slightly confusing. Wouldn't it be more consistent as `init_from_file_selection`, and `file_selection`?

I think it's important to keep consistency because the ability to represent cancellation is moved from being a `bool` flag to being a `None` option one level higher. At least if I'm reading this correctly.